src/Domain/UseCase/VerifyMFA.ts

```diff
@@ -26,32 +30,40 @@ export class VerifyMFA implements UseCaseInterface {
       }
     }
 
-    const mfaExtension = await this.itemsRepository.findMFAExtensionByUserUuid(user.uuid)
+    const mfaSecretSetting = await this.settingRepository.findOneByNameAndUserUuid(SETTINGS.MFA_SECRET, user.uuid)
 
-    if (!mfaExtension) {
+    if (!mfaSecretSetting) {
       return {
         success: true
       }
     }
 
-    const mfaParamKey = `mfa_${mfaExtension.uuid}`
-    if (!dto.requestParams[mfaParamKey]) {
+    if (!dto.token) {
       return {
         success: false,
         errorTag: 'mfa-required',
-        errorMessage: 'Please enter your two-factor authentication code.',
-        errorPayload: { mfa_key: mfaParamKey }
+        errorMessage: 'Please enter your two-factor authentication code.'
       }
     }
 
-    const mfaContent = this.contentDecoder.decode(mfaExtension.content)
+    const decryptedUserServerKey = await this.userServerKeyDecrypter.decrypt(user)
+    const [ version, mfaKey, mfaNonce ] = mfaSecretSetting.value.split(':')
```

Comment:
`mfaKey` should really be called either `encryptedMfaKey` or `ciphertext`, otherwise, that variable name is lying about what it is.